x-pack/plugins/reporting/server/export_types/csv_searchsource/generate_csv/generate_csv.test.ts

```diff
@@ -123,14 +108,26 @@ beforeEach(async () => {
     })
   );
 
-  searchSourceMock.getField = mockSearchSourceGetFieldDefault;
+  searchSourceMock.getField = jest.fn().mockImplementation((key: string) => {
+    switch (key) {
+      case 'index':
+        return {
+          fields: {
+            getByName: jest.fn().mockImplementation(() => []),
+            getByType: jest.fn().mockImplementation(() => []),
+          },
+          metaFields: ['_id', '_index', '_type', '_score'],
+          getFormatterForField: jest.fn(),
+        };
+    }
+  });
 });
 
 const logger = createMockLevelLogger();
 
 it('formats an empty search result to CSV content', async () => {
   const generateCsv = new CsvGenerator(
-    createMockJob({}),
+    createMockJob({ columns: ['date', 'ip', 'message'] }),
```

Comment:
All of the mock jobs in the test had to be updated to explicitly set the set of columns in the parameters. Previously, columns could be empty since it was assumed that the fields were set into the SearchSource in the browser. 

This PR does set the fields into the SearchSource in the browser, but not for most use cases. If Reporting tries to get its CSV columns using `searchSource.getField('fields')`, most likely it will not get usable field names.